inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -141,16 +277,31 @@ IE::Parameter AutoInferencePlugin::GetMetric(const std::string& name,
         std::vector<std::string> configKeys;
         IE_SET_METRIC_RETURN(SUPPORTED_CONFIG_KEYS, configKeys);
     } else if (name == METRIC_KEY(OPTIMIZATION_CAPABILITIES)) {
-        std::vector<std::string> capabilities = { "" };
+        std::vector<std::string> capabilities = GetOptimizationCapabilities();
         IE_SET_METRIC_RETURN(OPTIMIZATION_CAPABILITIES, capabilities);
     } else {
         IE_THROW() << "Unsupported metric key " << name;
     }
 }
 
+//////////////////////////////////// private & protected functions ///////////////////
 std::vector<AutoPlugin::DeviceInformation> AutoInferencePlugin::GetDeviceChoice(const ConfigType&  config) const {
     std::vector<DeviceInformation> metaDevices;
-    std::vector<std::string> availableDevices = GetCore()->GetAvailableDevices();
+    std::vector<std::string> availableDevices;
+
+    auto deviceListConfig = config.find(IE::AutoConfigParams::KEY_AUTO_DEVICE_LIST);
+    if (deviceListConfig == config.end()) {
+        availableDevices = GetCore()->GetAvailableDevices();
+    } else {
+        auto& devices = deviceListConfig->second;
+        std::string::size_type i = 0;
+        std::string::size_type idelimeter;
+        while ((idelimeter = devices.find(',', i)) != std::string::npos) {
+            availableDevices.push_back(devices.substr(i, idelimeter - i));
+            i = idelimeter + 1;
+        }
+        availableDevices.push_back(devices.substr(i, devices.length() - i));
```

Comment:
can we reuse `DeviceIDParser::getHeteroDevices` here?

Reply:
yes, I will change it